src/librustc_errors/emitter.rs

```diff
@@ -24,7 +24,23 @@ const ANONYMIZED_LINE_NUM: &str = "LL";
 /// Emitter trait for emitting errors.
 pub trait Emitter {
     /// Emit a structured diagnostic.
-    fn emit(&mut self, db: &DiagnosticBuilder<'_>);
+    fn emit_diagnostic(&mut self, db: &DiagnosticBuilder<'_>);
+
+    /// Emit a directive. The default is to do nothing; this should only be
+    /// produced with --error-format=json.
+    //
+    // njn: is "directive" a good name here? "json_directive"?
+    // "build_directive"?
+    //
+    // njn: is a string reasonable? Have an enum instead?
+    //
+    // njn: and now that Emitter emits not just errors but other messages, lots
```

Comment:
While true, not necessarily the end of the world. Just means it may take awhile to get everything updated!